From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Kleinschmidt <jkleinsc@github.com>
Date: Mon, 19 Nov 2018 18:33:56 -0500
Subject: Do not run arm/arm64 mksnapshot binaries

For arm and arm64 target_arches, Chromium builds mksnapshot as an x64 binary and
as part of that build mksnapshot is executed to produce snapshot_blob.bin.
Chromium does not build native arm and arm64 binaries of mksnapshot, but
Electron does, so this patch makes sure that the build doesn't try to run
the mksnapshot binary if it was built for arm or arm64.

diff --git a/BUILD.gn b/BUILD.gn
index 698b6c827f7f3cd7dc6871156daf8467bc7ff605..022c55b3fe32bbde57e97f08ab2c374cc1ea047a 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1240,9 +1240,19 @@ if (v8_use_snapshot && v8_use_external_startup_data) {
     ]
     public_deps = [
       ":natives_blob",
-      ":run_mksnapshot_default",
     ]
 
+    if (v8_snapshot_toolchain == "//build/toolchain/linux:clang_arm" ||
+        v8_snapshot_toolchain == "//build/toolchain/linux:clang_arm64") {
+      public_deps += [
+        ":mksnapshot($v8_snapshot_toolchain)",
+      ]
+    } else {
+      public_deps += [
+        ":run_mksnapshot_default",
+      ]
+    }
+
     if (v8_use_multi_snapshots) {
       public_deps += [ ":run_mksnapshot_trusted" ]
     }
